### ISSUE TITLE
Publish Docker image

### ISSUE THREAD

**Issue by terinjokes** [cloudflare/origin-ca-issuer#1]
Now that we're open source (🥳), it's time to get an image pushed to Docker Hub, and the deployment manifests updated to match.

**Comment by xunholy** [comment]
Please ensure the docker image is multi-architecture supported. Popular architectures like `arm64` often missed, using `buildx` should make supporting multiple architectures trivial, happy to help if needed.

**Comment by terinjokes** [comment]
The one built right now, that isn't being published anywhere, isn't multi-architecture. However, I agree, and hope to get that working soon.

**Comment by xunholy** [comment]
@terinjokes if you publish the Dockerfile and are happy I can set up some workflows in GHA and get it built and compiled for multi-architecture?

**Comment by terinjokes** [comment]
An update here: I'm still planning on doing this. I'm running into a temporary permissions issue in Docker Hub creating this image and getting credentials to publish from Actions. I hope to have this sorted soon. I'm really sorry for the delay.

**Comment by xunholy** [comment]
All good, can help out if required as I've done this many times, but let me know.

**Comment by terinjokes** [comment]
This isn't a technical problem, unfortunately.

**Comment by kg-ops** [comment]
Has this image been published yet ?

**Comment by terinjokes** [comment]
Nope. I still do not have the correct permissions in Docker Hub.

**Comment by kg-ops** [comment]
Is there a way to locally build this image ?

**Comment by terinjokes** [comment]
If you have nix installed you can build `./hack/docker.nix`. This kicks off the same image build [as GitHub Actions](https://github.com/cloudflare/origin-ca-issuer/blob/c821dd5a42890a575d42b0a808f8346a7985d579/.github/workflows/docker.yaml#L11).

```console
$ nix build -f ./hack/docker.nix
[11 built, 10 copied (1.5 MiB), 0.4 MiB DL]

$ docker load < result
docker load < result
a630bb07cd03: Loading layer [==================================================>]  1.649MB/1.649MB
14f0fd68d210: Loading layer [==================================================>]    256kB/256kB
7862e9b9223d: Loading layer [==================================================>]  31.63MB/31.63MB
ca397a15dfa0: Loading layer [==================================================>]  573.4kB/573.4kB
542652079767: Loading layer [==================================================>]  112.6kB/112.6kB
255355ea7603: Loading layer [==================================================>]  2.888MB/2.888MB
48ca96316146: Loading layer [==================================================>]  36.37MB/36.37MB
bab8c8268779: Loading layer [==================================================>]  10.24kB/10.24kB
Loaded image: origin-ca-issuer:dvx0mk1aysi6s46r2k3iis92b19klnk1
```

If you don't have nix installed, I've found the [nixpkgs/nix:latest](https://hub.docker.com/r/nixpkgs/nix) image works well, just make sure to copy the destination of the `result` symlink before exiting the container.

```console
$ docker -v $(pwd):/opt/work -w /opt/work --rm -it nixpkgs/nix
bash-4.4# nix build -f ./hack/docker.nix
[12 built, 137 copied (951.8 MiB), 225.9 MiB DL]

bash-4.4# cp result image.tar.gz

bash-4.4# exit

$ docker load < image.tar.gz
f04b2eecad2b: Loading layer [==================================================>]   36.4MB/36.4MB
bab8c8268779: Loading layer [==================================================>]  10.24kB/10.24kB
Loaded image: origin-ca-issuer:dvx0mk1aysi6s46r2k3iis92b19klnk1
```

Obviously, once I have access to pushing to Docker Hub you'll be able to `docker pull` (or, more relevantly, stick in your YAML) without needing to know or worry about nix at all.

---

If you don't feel like doing that, this is my local build from rev e06fedae20300b4552aaed60a6b2684a1d5964b8: [image.tar.gz](https://github.com/cloudflare/origin-ca-issuer/files/5657783/image.tar.gz) (21MB)

```console
$ sha256sum image.tar.gz 
ecd572a4402b771e01b7c8865142c737fc0992cd0cca598262878fc309e9c65c  image.tar.gz
```

**Comment by matthewdaltonditto** [comment]
what is blocking this?

**Comment by terinjokes** [comment]
I don't have permission to publish to the Cloudflare Docker Hub organization. I'm working with an internal team to get the permissions sorted out.

**Comment by terinjokes** [comment]
Sorry for the delay. I've published [cloudflare/origin-ca-issuer:v0.5.0](https://hub.docker.com/layers/cloudflare/origin-ca-issuer/v0.5.0/images/sha256-6f4520072e3cd14a5a863b9c478d291751d31876603722bfe70a9922aa6c4670?context=explore) for amd64 and arm64.
